fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/Literal.java

```diff
@@ -396,4 +400,153 @@ public boolean isZero() {
+            */
+            if (maxLen < 9) {
+                return 0;
+            }
+            len = MysqlProto.readInt4(data);
+            MysqlProto.readFixedString(data, 4);
+            return len;
+        } else if (len == 255) {
+            return 0;
+        } else {
+            return len;
+        }
+    }
+
+    /**
+     * Retrieves a Literal object based on the MySQL type and the data provided.
+     *
+     * @param mysqlType the MySQL type identifier
+     * @param data      the ByteBuffer containing the data
+     * @return a Literal object corresponding to the MySQL type
+     * @throws AnalysisException if the MySQL type is unsupported or if data conversion fails
+     */
+    public static Literal getLiteralByMysqlType(MysqlColType mysqlType, ByteBuffer data) throws AnalysisException {
+        switch (mysqlType) {
+            case MYSQL_TYPE_TINY:
+                return new TinyIntLiteral(data.get());
+            case MYSQL_TYPE_SHORT:
+                return new SmallIntLiteral((short) data.getChar());
+            case MYSQL_TYPE_LONG:
+                return new IntegerLiteral(data.getInt());
+            case MYSQL_TYPE_LONGLONG:
+                return new BigIntLiteral(data.getLong());
+            case MYSQL_TYPE_FLOAT:
+                return new FloatLiteral(data.getFloat());
+            case MYSQL_TYPE_DOUBLE:
+                return new DoubleLiteral(data.getDouble());
+            case MYSQL_TYPE_DECIMAL:
+            case MYSQL_TYPE_NEWDECIMAL:
+                return handleDecimalLiteral(data);
+            case MYSQL_TYPE_DATE:
+                return handleDateLiteral(data);
+            case MYSQL_TYPE_DATETIME:
+            case MYSQL_TYPE_TIMESTAMP:
+            case MYSQL_TYPE_TIMESTAMP2:
+                return handleDateTimeLiteral(data);
+            case MYSQL_TYPE_STRING:
+            case MYSQL_TYPE_VARSTRING:
+                return handleStringLiteral(data);
+            case MYSQL_TYPE_VARCHAR:
+                return handleVarcharLiteral(data);
+            default:
+                throw new AnalysisException("Unsupported MySQL type: " + mysqlType);
+        }
+    }
+
+    private static DecimalLiteral handleDecimalLiteral(ByteBuffer data) throws AnalysisException {
+        int len = getParmLen(data);
+        byte[] bytes = new byte[len];
+        data.get(bytes);
+        try {
+            String value = new String(bytes);
+            BigDecimal v = new BigDecimal(value);
+            return new DecimalLiteral(v);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid decimal literal", e);
+        }
+    }
+
+    private static DateLiteral handleDateLiteral(ByteBuffer data) {
+        int len = getParmLen(data);
+        if (len >= 4) {
+            int year = (int) data.getChar();
+            int month = (int) data.get();
+            int day = (int) data.get();
+            return new DateLiteral(year, month, day);
```

Comment:
should use `DateV2Literal` if Config.enable_date_conversion == true

Reply:
done